php/src/GildedRose.php

```diff
@@ -4,66 +4,34 @@
 
 namespace GildedRose;
 
+use ReflectionClass;
+
 final class GildedRose
 {
     /**
      * @var Item[]
      */
     private $items;
 
+    private $class = [
+        'normal' =>  "GildedRose\\normal",
```

Comment:
用字串組 namespace & class 的風險
```suggestion
        'normal' =>  "GildedRose\\Normal",
```